cmd/bosun/sched/sched.go

```diff
@@ -588,11 +598,38 @@ func (s *Schedule) action(user, message string, t models.ActionType, st *models.
 			return "", fmt.Errorf("cannot acknowledge closed alert")
 		}
 		st.NeedAck = false
+	case models.ActionCancelClose:
+		found := false
+		for i, a := range st.Actions {
+			// Find first delayed close that hasn't already been fulfilled or canceled
+			if a.Type == models.ActionDelayedClose && !(a.Fullfilled || a.Cancelled) {
+				found, st.Actions[i].Cancelled = true, true
+				break
+			}
+		}
+		if !found {
+			return "", fmt.Errorf("no delayed close for incident %v (%v) found to cancel", st.Id, st.AlertKey)
+		}
 	case models.ActionClose:
-		if st.IsActive() {
-			return "", fmt.Errorf("cannot close active alert")
+		// closing effectively acks the incident
+		st.NeedAck = false
+		if st.IsActive() { // Closing an active incident results in delayed close
+			action.Type = models.ActionDelayedClose
+			alertDef := s.RuleConf.GetAlert(st.AlertKey.Name())
+			if at != nil {
+				action.Deadline = at
+			} else { // Set deadline based on Alert Check Frequency * 2
+				runEvery := alertDef.RunEvery
+				if runEvery == 0 {
```

Comment:
alertDef.RunEvery should be set for each alert at startup time. Or is there a helper function? In any case, we should not repeat that here.